packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -0,0 +1,339 @@
+  });
+
+  const priceBook = provide(baggage, 'sortedOffers', () => {
+    const priceStore = makeScalarBigMapStore('sortedOffersStore', {
+      durable: true,
+    });
+    return makePriceBook(priceStore, currencyBrand, collateralBrand);
+  });
+
+  const removeFromOneBook = (isPriceBook, key) => {
+    if (isPriceBook) {
+      priceBook.delete(key);
+    } else {
+      scaledBidBook.delete(key);
+    }
+  };
+
+  // Settle with seat. The caller is responsible for updating the book, if any.
+  const settle = (seat, collateralWanted) => {
+    const { Currency: currencyAvailable } = seat.getCurrentAllocation();
+    const { Collateral: collateralAvailable } =
+      collateralSeat.getCurrentAllocation();
+    if (!collateralAvailable || AmountMath.isEmpty(collateralAvailable)) {
+      return AmountMath.makeEmptyFromAmount(collateralWanted);
+    }
+
+    /** @type {Amount<'nat'>} */
+    const collateralTarget = AmountMath.min(
+      collateralWanted,
+      collateralAvailable,
+    );
+
+    const currencyNeeded = ceilMultiplyBy(collateralTarget, curAuctionPrice);
+    if (AmountMath.isEmpty(currencyNeeded)) {
+      seat.fail('price fell to zero');
+      return AmountMath.makeEmptyFromAmount(collateralWanted);
+    }
+
+    let collateralRecord;
+    let currencyRecord;
+    if (AmountMath.isGTE(currencyAvailable, currencyNeeded)) {
+      collateralRecord = {
+        Collateral: collateralTarget,
+      };
+      currencyRecord = {
+        Currency: currencyNeeded,
+      };
+    } else {
+      const affordableCollateral = floorDivideBy(
+        currencyAvailable,
+        curAuctionPrice,
+      );
+      collateralRecord = {
+        Collateral: affordableCollateral,
+      };
+      currencyRecord = {
+        Currency: currencyAvailable,
+      };
+    }
+
+    trace('settle', { currencyRecord, collateralRecord });
+
+    atomicRearrange(
+      zcf,
+      harden([
+        [collateralSeat, seat, collateralRecord],
+        [seat, currencySeat, currencyRecord],
+      ]),
+    );
+    return collateralRecord.Collateral;
+  };
+
+  const isActive = auctionState => auctionState === AuctionState.ACTIVE;
+
+  /**
+   *  Accept an offer expressed as a price. If the auction is active, attempt to
+   *  buy collateral. If any of the offer remains add it to the book.
+   *
+   *  @param {ZCFSeat} seat
+   *  @param {Ratio} price
+   *  @param {Amount} want
+   *  @param {AuctionState} auctionState
+   */
+  const acceptPriceOffer = (seat, price, want, auctionState) => {
+    trace('acceptPrice');
+    // Offer has ZcfSeat, offerArgs (w/price) and timeStamp
+
+    let collateralSold = AmountMath.makeEmptyFromAmount(want);
+    if (isActive(auctionState) && ratioGTE(price, curAuctionPrice)) {
+      collateralSold = settle(seat, want);
+
+      if (AmountMath.isEmpty(seat.getCurrentAllocation().Currency)) {
+        seat.exit();
+        return;
+      }
+    }
```

Comment:
the double nested return condition is a little hard to follow. is this equivalent?
```suggestion
    const collateralSold = isActive(auctionState) && ratioGTE(price, curAuctionPrice)
      ? settle(seat, want)
      : AmountMath.makeEmptyFromAmount(want);
      
    if (AmountMath.isEmpty(seat.getCurrentAllocation().Currency)) {
      seat.exit();
      return;
    }
```

Reply:
Good suggestion. The result is even simpler, I think.